api/queue/work/model_service_deployment_test.go

```diff
@@ -488,11 +601,16 @@ func TestExecuteDeployment(t *testing.T) {
 				assert.Equal(t, env.DefaultResourceRequest, savedEndpoint.ResourceRequest)
 			}
 
+			mockDeploymentStorage.AssertNumberOfCalls(t, "GetLatestDeployment", 1)
 			if tt.deployErr != nil {
 				mockDeploymentStorage.AssertNumberOfCalls(t, "Save", 2)
 				assert.Equal(t, models.EndpointFailed, savedEndpoint.Status)
 			} else {
-				mockDeploymentStorage.AssertNumberOfCalls(t, "Save", 1)
+				if tt.endpoint.Status == models.EndpointPending {
```

Comment:
This condition checks if an endpoint is originally in the 'pending' state - if it is, we will expect the deployment attempt to not bother creating a new deployment entry in the deployments table, hence the 0 calls expected for the number of db save operations.